### PR TITLE
fix(turborepo): Better unlink Output

### PR DESCRIPTION
**Pull request by nathanhammond**
`turbo unlink` previously would delete the entire configuration file which is unnecessarily punishing, this switches to do the absolute minimum change to the file.

Closes TURBO-1225

### CODE REVIEW

File: crates/turborepo-lib/src/commands/unlink.rs [hunk 1]
```diff
@@ -11,26 +11,48 @@ enum UnlinkSpacesResult {
...
-        .context("could not unlink. Something went wrong")?;
+    let repo_config: &mut crate::config::RepoConfig = base.repo_config_mut()?;
+    let needs_disabling = repo_config.team_id().is_some() || repo_config.team_slug().is_some();
 
-    println!(
-        "{}",
-        base.ui.apply(GREY.apply_to("> Disabled Remote Caching"))
-    );
+    if needs_disabling {
+        base.repo_config_mut()?.set_team_id(None)?;
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/commands/unlink.rs:18]: Can we not reuse `repo_config` here?
```suggestion
        repo_config.set_team_id(None)?;
```

File: crates/turborepo-lib/src/commands/unlink.rs [hunk 2]
```diff
@@ -11,26 +11,48 @@ enum UnlinkSpacesResult {
 }
 
 fn unlink_remote_caching(base: &mut CommandBase) -> Result<()> {
-    base.delete_repo_config_file()
-        .context("could not unlink. Something went wrong")?;
+    let repo_config: &mut crate::config::RepoConfig = base.repo_config_mut()?;
+    let needs_disabling = repo_config.team_id().is_some() || repo_config.team_slug().is_some();
```
Comment (chris-olszewski) [inline crates/turborepo-lib/src/commands/unlink.rs:15]: Would it be worth moving this logic into the config struct itself to something like `maybe_disable_remote_caching -> Result<bool, Error>`?

**Review by chris-olszewski — APPROVED** [review]

File: crates/turborepo-lib/src/commands/unlink.rs (hunk 2 above)
Comment (nathanhammond) [reply crates/turborepo-lib/src/commands/unlink.rs:15]: I _almost_ did this, but I want to remove `team_slug` entirely and change the shape of how this works by default. It felt premature.

File: crates/turborepo-lib/src/commands/unlink.rs (hunk 1 above)
Comment (nathanhammond) [reply crates/turborepo-lib/src/commands/unlink.rs:18]: I did this in isolation from the `*_config` handling change; this is a good catch.

File: crates/turborepo-lib/src/commands/unlink.rs [hunk 3]
```diff
@@ -11,26 +11,48 @@ enum UnlinkSpacesResult {
...
+            "{}",
+            base.ui.apply(GREY.apply_to("> Disabled Remote Caching"))
+        );
+    } else {
+        println!(
+            "{}",
+            base.ui
+                .apply(GREY.apply_to("> No Remote Caching config found"))
+        );
+    }
```
[the remark this answers is not in the document]
Comment (nathanhammond) [reply crates/turborepo-lib/src/commands/unlink.rs:30]: I decided to understand this as "the visual noise of the logging makes it hard to tell what is going on."

New pattern:
```rust
let output;
if needs_disabling {
    repo_config.set_team_id(None)?;
    output = "> Disabled Remote Caching";
} else {
    output = "> No Remote Caching config found";
}

println!(
    "{}",
    base.ui.apply(GREY.apply_to(output))
);
```

**Pull request merged**